packages/valory/skills/decision_maker_abci/behaviours/base.py

```diff
@@ -85,6 +89,25 @@ def __init__(self, **kwargs: Any) -> None:
         self.multisend_data = b""
         self._safe_tx_hash = ""
         self._policy: Optional[EGreedyPolicy] = None
+        self._inflight_strategy_req: Optional[str] = None
+
+    @property
+    def strategy_exec(self) -> str:
+        """Get the executable strategy file's content."""
+        return self.shared_state.strategies_executables[self.params.trading_strategy]
+
+    def execute_strategy(self, *args: Any, **kwargs: Any) -> Dict[str, Any]:
+        """Execute the strategy and return the results."""
+        if STRATEGY_RUN_METHOD in globals():
+            del globals()[STRATEGY_RUN_METHOD]
+        exec(self.strategy_exec, globals())  # pylint: disable=W0122  # nosec
+        method = globals().get(STRATEGY_RUN_METHOD, None)
+        if method is None:
+            self.context.logger.error(
+                f"No {STRATEGY_RUN_METHOD!r} method was found in {self.params.trading_strategy} strategy's executable."
+            )
+            return {BET_AMOUNT_FIELD: 0}
+        return method(*args, **kwargs)
```

Comment:
So basically the assumption here is that the strategy doesnt block. I guess its fine for now, although maybe it makes sense to run this on a separate thread/process to not block the agent.

Reply:
Opened #169.